icu4j/main/classes/core/src/com/ibm/icu/impl/units/ComplexUnitsConverter.java

```diff
@@ -115,6 +115,9 @@ public boolean greaterThanOrEqual(BigDecimal quantity, BigDecimal limit) {
                 // Keep the residual of the quantity.
                 //   For example: `3.6 feet`, keep only `0.6 feet`
                 quantity = quantity.subtract(newQuantity);
+                if (quantity.compareTo(BigDecimal.ZERO) == -1) {
+                    quantity = BigDecimal.ZERO;
+                }
```

Comment:
For the future: given the C++ version zeroes not just negative numbers but also "+epsilon", I want to see if I can write a unit test which will output something that's just a smidge above zero.

            // Keep the residual of the quantity.
            //   For example: `3.6 feet`, keep only `0.6 feet`
            //
            // When the calculation is near enough +/- DBL_EPSILON, we round to
            // zero. (We also ensure no negative values here.)
            if ((quantity - roundedQuantity) / quantity < DBL_EPSILON) {
                quantity = 0;
            } else {
                quantity -= roundedQuantity;
            }